recipes/botan/all/conanfile.py

```diff
@@ -0,0 +1,320 @@
+
+    @property
+    def _configure_cmd(self):
+        if self.settings.compiler in ('clang', 'apple-clang'):
+            botan_compiler = 'clang'
+        elif self.settings.compiler == 'gcc':
+            botan_compiler = 'gcc'
+        else:
+            botan_compiler = 'msvc'
+
+        botan_abi_flags = []
+        botan_extra_cxx_flags = []
+        build_flags = []
+
+        if self._is_linux_clang_libcxx:
+            botan_abi_flags.extend(["-stdlib=libc++", "-lc++abi"])
+
+        if botan_compiler in ['clang', 'apple-clang', 'gcc']:
+            if self.settings.arch == "x86":
+                botan_abi_flags.append('-m32')
+            elif self.settings.arch == "x86_64":
+                botan_abi_flags.append('-m64')
+
+        if self.settings.os != "Windows" and self.options.fPIC:
+            botan_extra_cxx_flags.append('-fPIC')
+
+        # This is to work around botan's configure script that *replaces* its
+        # standard (platform dependent) flags in presence of an environment
+        # variable ${CXXFLAGS}. Most notably, this would build botan with
+        # disabled compiler optimizations.
+        environment_cxxflags = tools.get_env("CXXFLAGS")
+        if environment_cxxflags:
+            del os.environ["CXXFLAGS"]
+            botan_extra_cxx_flags.append(environment_cxxflags)
+
+        if self.options.enable_modules:
+            build_flags.append('--minimized-build')
+            build_flags.append('--enable-modules={}'.format(self.options.enable_modules))
+
+        if self.options.amalgamation:
+            build_flags.append('--amalgamation')
+
+        if self.options.single_amalgamation:
+            build_flags.append('--single-amalgamation-file')
+
+        if self.options.system_cert_bundle:
+            build_flags.append('--system-cert-bundle={}'.format(self.options.system_cert_bundle))
+
+        if self.options.bzip2:
+            build_flags.append('--with-bzip2')
+            build_flags.extend(self._dependency_build_flags("bzip2"))
+
+        if self.options.openssl:
+            build_flags.append('--with-openssl')
+            build_flags.extend(self._dependency_build_flags("OpenSSL"))
+
+        if self.options.quiet:
+            build_flags.append('--quiet')
+
+        if self.options.sqlite3:
+            build_flags.append('--with-sqlite3')
+            build_flags.extend(self._dependency_build_flags("sqlite3"))
+
+        if self.options.zlib:
+            build_flags.append('--with-zlib')
+            build_flags.extend(self._dependency_build_flags("zlib"))
+
+        if self.options.boost:
+            build_flags.append('--with-boost')
+            build_flags.extend(self._dependency_build_flags("boost"))
+            # required boost libraries are listed in Botan's src/utils/boost/info.txt
+            # under the <libs></libs> tag...
+            # Note that boost_system is actually a header-only library as of
```

Comment:
Since this is using Boost 1.70 this seems unnecessary. 
In my opinion, we shouldn't add workarounds for the case that someone is overwriting a dependency